src/app/shop/[category]/_components/TitleWrap/Filter.tsx

```diff
@@ -5,31 +5,109 @@ import { FILTER_OPTIONS } from '@/constants/product';
 import classNames from 'classnames/bind';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import queryString from 'query-string';
 import { useState } from 'react';
 import styles from './Filter.module.scss';
 
 const cn = classNames.bind(styles);
 
 const DELETE_ITEM_COLOR = '#787878';
 
-export default function Filter({ category }: { category: CategoryKey }) {
-  const [selectedList, setSelectedList] = useState<string[]>([]);
+interface FilterProps {
+  category: CategoryKey;
+}
+
+export default function Filter({ category }: FilterProps) {
+  const searchParams = useSearchParams();
+  const [selectedManufacturers, setSelectedManufacturers] = useState<string[]>(
+    searchParams.get('companies')?.split(',') || [],
+  );
+  const [selectedSwitchTypes, setSelectedSwitchTypes] = useState<string[]>(
+    searchParams.get('switchTypes')?.split(',') || [],
+  );
+  const [minPriceState, setMinPriceState] = useState<string>(searchParams.get('minPrice') || '');
+  const [maxPriceState, setMaxPriceState] = useState<string>(searchParams.get('maxPrice') || '');
+  const [tempMinPrice, setTempMinPrice] = useState<string>(minPriceState);
+  const [tempMaxPrice, setTempMaxPrice] = useState<string>(maxPriceState);
+
   const options = FILTER_OPTIONS[category];
+  const pathname = usePathname();
+  const router = useRouter();
+
+  const handleSearch = (companies: string[], switchTypes: string[], minPrice: string, maxPrice: string) => {
+    const query: { [key: string]: string | undefined } = {
+      companies: companies.length ? companies.join(',') : undefined,
+      switchTypes: switchTypes.length ? switchTypes.join(',') : undefined,
+      minPrice: minPrice || undefined,
+      maxPrice: maxPrice || undefined,
+    };
+
+    router.push(
+      queryString.stringifyUrl(
+        {
+          url: pathname,
+          query,
+        },
+        { skipNull: true, skipEmptyString: true },
+      ),
+    );
+  };
 
   const handleClickReset = () => {
-    setSelectedList([]);
+    setSelectedManufacturers([]);
+    setSelectedSwitchTypes([]);
+    setMinPriceState('');
+    setMaxPriceState('');
+    setTempMinPrice('');
+    setTempMaxPrice('');
+    handleSearch([], [], '', '');
+  };
+
+  const handleClickFilterItem = (item: string, type: 'manufacturer' | 'switch') => {
+    if (type === 'manufacturer') {
+      setSelectedManufacturers((prev) => {
+        const newSelected = prev.includes(item) ? prev.filter((x) => x !== item) : [...prev, item];
+        handleSearch(newSelected, selectedSwitchTypes, minPriceState, maxPriceState);
+        return newSelected;
+      });
+    } else if (type === 'switch') {
+      setSelectedSwitchTypes((prev) => {
+        const newSelected = prev.includes(item) ? prev.filter((x) => x !== item) : [...prev, item];
+        handleSearch(selectedManufacturers, newSelected, minPriceState, maxPriceState);
+        return newSelected;
+      });
```

Comment:
handleSearch로 불러오는 값만 달라서 아래처럼도 할 수 있을 것 같아요

```suggestion
      setSelectedManufacturers((prev) => {
        const newSelected = prev.includes(item) ? prev.filter((x) => x !== item) : [...prev, item];
        type === 'manufactiurer' 
        ? handleSearch(newSelected, selectedSwitchTypes, minPriceState, maxPriceState)
         : handleSearch(selectedManufacturers, newSelected, minPriceState, maxPriceState);
        return newSelected;
      });
```
그런데 setState함수 내부에 router.push가 들어간 함수를 실행하는 게 맞는지 살짝 의문이네요.